### ISSUE TITLE
Dependency automation for supported module types

### ISSUE THREAD

**Issue by calvincestari** [apollographql/apollo-ios#2034]
This task is one of the final steps in code generation; to take the generated schema files and package them together in the required [dependency manager format](https://github.com/apollographql/apollo-ios/blob/release/1.0-alpha-incubating/Sources/ApolloCodegenLib/ApolloCodegenConfiguration.swift#L100).

- [x] Swift Package Manager
- [x] None (previously "Manually Linked")
- [x] Other (previously "Carthage" and "CocoaPods")
- [ ] Swift Package Manager (snippet - TBD)

**Comment by calvincestari** [comment]
Sequencing could be an issue re. having this package built before any other package manager build begins.

**Comment by calvincestari** [comment]
I'm marking this issue as "Todo" again as SPM support is all we're going to do for the first alpha release. Supporting the other dependency managers will come afterwards.

**Comment by calvincestari** [comment]
A bit more detail on how we're changing support for dependency managers and the generated schema module.

Initially we decided to support named implementations for Swift Package Manager, CocoaPods and Carthage. SPM is easy enough since the Package.swift file is fairly concise and we can generate all required metadata with what we know about the schema module.

CocoaPods makes this more complicated with the [podspec format](https://guides.cocoapods.org/syntax/podspec.html#group_root_specification) which would require us to gather a lot more detail from the user for the metadata. Carthage takes this a step further and [requires an Xcode project file](https://github.com/Carthage/Carthage#differences-between-carthage-and-cocoapods). 😢 

Instead of growing the already large codegen configuration struct we've opted to support 'other' dependency managers by requiring the user to create the infrastructure needed, such as the podspec for CocoaPods or the Xcode project for Carthage. Combine that with the generated schema module files output to a location of the user's choosing and we can support a much wider range of dependency managers than we can write specific implementations for.
